### PR TITLE
[FIX] web: aggregate cells are not formatted with the monetary digits precision

### PR DESCRIPTION
**Pull request by hubvd**
**Steps to follow**

  - Use a currency with more than 2 digits of precision
  - Create a journal entry
  - Add a few lines
  -> The sum is not formatted to the correct precision

**Cause of the issue**

  The currency is not taken into account when calling formatMonetary

opw-2753312
task-2822174

### CODE REVIEW

**Review by sdegueldre — CHANGES_REQUESTED** [review]
Missing a test

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 1]
```diff
@@ -450,9 +451,21 @@ var ListRenderer = BasicRenderer.extend({
...
+                if (field.type === 'monetary' && !digits) {
+                    const currencyField = field.currency_field && self.state.fieldsInfo.list[field.currency_field];
+                    const currency_id = currencyField && self.state.data.length &&
+                        self.state.data[0].data[field.currency_field].res_id;
+
+                    if (currency_id) {
+                        const currency = session.get_currency(currency_id);
+                        digits = currency.digits;
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (hubvd) [reply addons/web/static/src/js/views/list/list_renderer.js:464]: What if I format with the currency only if all records share the same currency ?

**Comment by hubvd** [comment]
Hey @aab-odoo,
I updated this PR according to the task. What do you think ?

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 2]
```diff
@@ -450,9 +451,28 @@ var ListRenderer = BasicRenderer.extend({
                 if (!formatFunc) {
                     formatFunc = field_utils.format[field.type];
                 }
+                let digits = column.attrs.digits ? JSON.parse(column.attrs.digits) : undefined;
+                if (field.type === 'monetary') {
+                    const currencyField = field.currency_field && self.state.fieldsInfo.list[field.currency_field];
+                    if (currencyField && self.state.data.length) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js:457]: self.state.date.length could be checked in the first if

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 3]
```diff
@@ -450,9 +451,28 @@ var ListRenderer = BasicRenderer.extend({
...
+                        const currencies = self.state.data.map(entry => entry.data[field.currency_field].res_id);
+                        const singleCurrency = currencies[0] && currencies.every(currency => currency === currencies[0]);
+                        if (singleCurrency && !digits) {
+                            const currency = session.get_currency(currencies[0]);
+                            digits = currency.digits;
+                        } else {
+                            return $cell.text('N/A').tooltip({
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js:464]: should be translatable -> _t("N/A")

File: addons/web/static/tests/views/list_tests.js [hunk 4]
```diff
@@ -2217,6 +2219,50 @@ QUnit.module('Views', {
...
+             symbol: '₽',
+             position: 'after',
+             rounding: 0.0001,
+             decimal_places: 4,
+         });
+
+         this.data.foo.records.forEach(e => e.currency_id = 3);
+
+         const currencies = {};
+         this.data.res_currency.records.forEach(currency => {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list_tests.js:2238]: you can modify data in place, each  test receives a deep copy

File: addons/web/static/tests/views/list_tests.js [hunk 5]
```diff
@@ -2217,6 +2219,50 @@ QUnit.module('Views', {
...
+
+        assert.strictEqual(list.$('.o_data_row td:nth(1)').text(), '1200.0000' + '\xa0' + '₽',
+            "field should still be formatted based on currency");
+        assert.strictEqual(list.$('tfoot td:nth(1)').text(), '2000.0000',
+            "aggregates monetary is formatted with the currency " +
+            "digits if no digits attributes has been set");
+
+        list.destroy();
+    });
+
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list_tests.js:2265]: there should be a test for the N/A case, and one for the case where there is a digits attrs on the column

**Review by aab-odoo — COMMENTED** [review]
It's not obvious to me that this should be done in v14. I would rather say this is an improvement for master. In any cases, I think the change is good functionally speaking. What matters the most here is the tests, as we are currently rewriting the list view from scratch. So having this new behavior fully tested is the only way to ensure that we won't break it directly. From that perspective, there's still a bit of work to do.

**Pull request closed without merging**